client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -399,17 +398,33 @@ function mediaButton( editor ) {
 		delete media.caption;
 		media = assign( {}, parsed.media, media );
 
+		// Compute a ratio used to order media sizes
+		const computeRatio = ( { width = media.width, height = media.height } ) => {
+			return Math.min(
+				( width / media.width ) || Infinity,
+				( height / media.height ) || Infinity
+			);
+		};
+
 		// Determine the next usable size
-		let sizeIndex = SIZE_ORDER.indexOf( parsed.appearance.size );
+		const sizeRatios = SIZE_ORDER
+			.map( size => MediaUtils.getThumbnailSizeDimensions( size, site ) )
+			.map( computeRatio );
```

Comment:
I hesitated and I would have done it at first if the array to loop on was bigger than that. 
I had to make a choice between clarity and perf, I've chosen clarity of code in this case, but I understand your point of view. Let me update this.